===== Patch instructions generated by SAVER =====
[+] { Insert: if (true) free(*(open_users_acl:filename)) at 32 (line 322, column 7) }
===== Actual patch in diff format =====
--- a/table1/benchmarks/inetutils-1.9.4/talkd/acl.c
+++ b/table1/benchmarks/inetutils-1.9.4/talkd/acl.c
@@ -320,6 +320,7 @@ open_users_acl (char *name)
    */
   rc = stat (filename, &st);
   if (rc < 0)
+    free(filename);
     return NULL;
   if (!S_ISREG(st.st_mode)
       || st.st_uid != pw->pw_uid
